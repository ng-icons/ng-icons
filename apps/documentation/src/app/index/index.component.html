<div
  appFadeInContainer
  class="flex flex-col items-center justify-center h-screen px-4 mt-6 md:mt-0"
>
  <img
    appFadeIn
    class="w-32 mb-6"
    src="assets/angular.svg"
    alt="Angular Logo"
  />

  <h2
    appFadeIn
    class="heading mb-5 text-base md:text-xl font-extrabold uppercase"
  >
    Angular Icons
  </h2>

  <h1
    appFadeIn
    class="slogan font-extrabold text-3xl md:text-5xl text-center mb-6 md:leading-[60px]"
  >
    The all-in-one icon<br />
    library for Angular
  </h1>

  <p appFadeIn class="text-sm md:text-base text-[#F5EAEA] text-center mb-16">
    Providing over 48,000 icons for use in your applications
  </p>

  <div appFadeIn class="flex gap-6 flex-col md:flex-row">
    <button
      routerLink="/getting-started"
      class="btn btn-white h-12 px-6 flex gap-x-2 items-center font-medium text-sm transition-shadow ease-out duration-200"
    >
      <span class="mb-0.5">Get Started</span>

      <ng-icon name="heroChevronRightMini" class="text-xl"></ng-icon>
    </button>

    <button
      routerLink="/browse-icons"
      class="btn btn-red h-12 px-6 flex gap-x-2 items-center font-medium text-sm transition-shadow ease-out duration-200"
    >
      <span class="mb-0.5">Browse Icons</span>

      <ng-icon name="heroChevronRightMini" class="text-xl"></ng-icon>
    </button>
  </div>
</div>
