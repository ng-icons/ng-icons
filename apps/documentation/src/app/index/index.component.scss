.heading {
  line-height: 25px;
  letter-spacing: 0.08em;
  background: linear-gradient(360deg, #ffafbd 0%, #ffc3a0 82%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-clip: text;
  text-fill-color: transparent;
}

.slogan {
  line-height: 60px;
  background: linear-gradient(
    180deg,
    #ffffff 0%,
    rgba(227, 153, 153, 0.96) 100%
  );
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-clip: text;
  text-fill-color: transparent;
}

.btn {
  border-radius: 32px;
  box-shadow: 0 0 0 rgba(0, 0, 0, 0);

  &.btn-white {
    background: linear-gradient(90deg, #e0eafc 0%, #cfdef3 100%);
    border-top: 1px solid #ffffff;
    color: #1a1e2a;

    &:hover {
      box-shadow: 0px 1px 40px rgba(224, 234, 252, 0.4);
    }
  }

  &.btn-red {
    border-top: 1px solid rgba(238, 238, 238, 0.3);
    background: linear-gradient(90deg, #e53935 0%, #e35d5b 100%);
    color: #fff;

    &:hover {
      box-shadow: 0px 1px 40px rgba(216, 24, 47, 0.5);
    }
  }
}
