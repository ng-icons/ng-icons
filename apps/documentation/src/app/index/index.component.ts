import { Component } from '@angular/core';
import { RouterLink } from '@angular/router';
import { NgIconComponent, provideIcons } from '@ng-icons/core';
import { heroChevronRightMini } from '@ng-icons/heroicons/mini';
import { FadeInContainerDirective } from '../directives/fade-in/fade-in-container.directive';
import { FadeInDirective } from '../directives/fade-in/fade-in.directive';

@Component({
  selector: 'app-index',
  templateUrl: './index.component.html',
  styleUrls: ['./index.component.scss'],
  standalone: true,
  imports: [
    NgIconComponent,
    RouterLink,
    FadeInContainerDirective,
    FadeInDirective,
  ],
  providers: [provideIcons({ heroChevronRightMini })],
})
export class IndexComponent {}
