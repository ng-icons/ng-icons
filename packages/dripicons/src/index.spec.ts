import * as icons from './index';

describe('Dripicons', () => {
  it('should ensure the export names have not changed unexpectedly', () => {
    expect(Object.keys(icons)).toMatchSnapshot();
  });
});
