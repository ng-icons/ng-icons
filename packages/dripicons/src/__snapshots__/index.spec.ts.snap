// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dripicons should ensure the export names have not changed unexpectedly 1`] = `
Array [
  "dripAlarm",
  "dripAlignCenter",
  "dripAlignJustify",
  "dripAlignLeft",
  "dripAlignRight",
  "dripAnchor",
  "dripArchive",
  "dripArrowDown",
  "dripArrowLeft",
  "dripArrowRight",
  "dripArrowThinDown",
  "dripArrowThinLeft",
  "dripArrowThinRight",
  "dripArrowThinUp",
  "dripArrowUp",
  "dripArticle",
  "dripBackspace",
  "dripBasket",
  "dripBasketball",
  "dripBatteryEmpty",
  "dripBatteryFull",
  "dripBatteryLow",
  "dripBatteryMedium",
  "dripBell",
  "dripBlog",
  "dripBluetooth",
  "dripBold",
  "dripBookmark",
  "dripBookmarks",
  "dripBox",
  "dripBriefcase",
  "dripBrightnessLow",
  "dripBrightnessMax",
  "dripBrightnessMedium",
  "dripBroadcast",
  "dripBrowserUpload",
  "dripBrowser",
  "dripBrush",
  "dripCalendar",
  "dripCamcorder",
  "dripCamera",
  "dripCard",
  "dripCart",
  "dripChecklist",
  "dripCheckmark",
  "dripChevronDown",
  "dripChevronLeft",
  "dripChevronRight",
  "dripChevronUp",
  "dripClipboard",
  "dripClock",
  "dripClockwise",
  "dripCloudDownload",
  "dripCloudUpload",
  "dripCloud",
  "dripCode",
  "dripContract2",
  "dripContract",
  "dripConversation",
  "dripCopy",
  "dripCrop",
  "dripCross",
  "dripCrosshair",
  "dripCutlery",
  "dripDeviceDesktop",
  "dripDeviceMobile",
  "dripDeviceTablet",
  "dripDirection",
  "dripDisc",
  "dripDocumentDelete",
  "dripDocumentEdit",
  "dripDocumentNew",
  "dripDocumentRemove",
  "dripDocument",
  "dripDot",
  "dripDots2",
  "dripDots3",
  "dripDownload",
  "dripDuplicate",
  "dripEnter",
  "dripExit",
  "dripExpand2",
  "dripExpand",
  "dripExperiment",
  "dripExport",
  "dripFeed",
  "dripFlag",
  "dripFlashlight",
  "dripFolderOpen",
  "dripFolder",
  "dripForward",
  "dripGaming",
  "dripGear",
  "dripGraduation",
  "dripGraphBar",
  "dripGraphLine",
  "dripGraphPie",
  "dripHeadset",
  "dripHeart",
  "dripHelp",
  "dripHome",
  "dripHourglass",
  "dripInbox",
  "dripInformation",
  "dripItalic",
  "dripJewel",
  "dripLifting",
  "dripLightbulb",
  "dripLinkBroken",
  "dripLink",
  "dripList",
  "dripLoading",
  "dripLocation",
  "dripLockOpen",
  "dripLock",
  "dripMail",
  "dripMap",
  "dripMediaLoop",
  "dripMediaNext",
  "dripMediaPause",
  "dripMediaPlay",
  "dripMediaPrevious",
  "dripMediaRecord",
  "dripMediaShuffle",
  "dripMediaStop",
  "dripMedical",
  "dripMenu",
  "dripMessage",
  "dripMeter",
  "dripMicrophone",
  "dripMinus",
  "dripMonitor",
  "dripMove",
  "dripMusic",
  "dripNetwork1",
  "dripNetwork2",
  "dripNetwork3",
  "dripNetwork4",
  "dripNetwork5",
  "dripPamphlet",
  "dripPaperclip",
  "dripPencil",
  "dripPhone",
  "dripPhotoGroup",
  "dripPhoto",
  "dripPill",
  "dripPin",
  "dripPlus",
  "dripPower",
  "dripPreview",
  "dripPrint",
  "dripPulse",
  "dripQuestion",
  "dripReplyAll",
  "dripReply",
  "dripReturn",
  "dripRetweet",
  "dripRocket",
  "dripScale",
  "dripSearch",
  "dripShoppingBag",
  "dripSkip",
  "dripStack",
  "dripStar",
  "dripStopwatch",
  "dripStore",
  "dripSuitcase",
  "dripSwap",
  "dripTagDelete",
  "dripTag",
  "dripTags",
  "dripThumbsDown",
  "dripThumbsUp",
  "dripTicket",
  "dripTimeReverse",
  "dripToDo",
  "dripToggles",
  "dripTrash",
  "dripTrophy",
  "dripUpload",
  "dripUserGroup",
  "dripUserId",
  "dripUser",
  "dripVibrate",
  "dripViewApps",
  "dripViewListLarge",
  "dripViewList",
  "dripViewThumb",
  "dripVolumeFull",
  "dripVolumeLow",
  "dripVolumeMedium",
  "dripVolumeOff",
  "dripWallet",
  "dripWarning",
  "dripWeb",
  "dripWeight",
  "dripWifi",
  "dripWrong",
  "dripZoomIn",
  "dripZoomOut",
]
`;
